src/HazelcastClient.ts

```diff
@@ -129,25 +130,102 @@ export class HazelcastClient {
-        this.nearCacheManager = new NearCacheManager(this);
-        this.partitionService = new PartitionServiceImpl(this);
-        this.lifecycleService = new LifecycleServiceImpl(this);
+        this.nearCacheManager = new NearCacheManager(
+            this.config,
+            this.serializationService
+        );
+        this.partitionService = new PartitionServiceImpl(
+            this.loggingService.getLogger(),
+            this.serializationService
+        );
         this.clusterFailoverService = this.initClusterFailoverService();
-        this.connectionManager = new ClientConnectionManager(this);
-        this.invocationService = new InvocationService(this);
-        this.proxyManager = new ProxyManager(this);
-        this.cpSubsystem = new CPSubsystemImpl(this);
-        this.clusterService = new ClusterService(this);
+        this.lifecycleService = new LifecycleServiceImpl(
+            this,
+            this.config,
+            this.loggingService.getLogger()
+        );
+
+        this.clusterService = new ClusterService(
+            this.config,
+            this.loggingService.getLogger(),
+            this.clusterFailoverService
+        );
+
+        this.invocationService = new InvocationService(
+            this.config,
+            this.loggingService.getLogger(),
+            this.partitionService,
+            this.errorFactory,
+            this.lifecycleService
+        );
+
+        this.connectionManager = new ClientConnectionManager(
+            this,
+            this.instanceName,
+            this.config,
+            this.loggingService.getLogger(),
+            this.partitionService,
+            this.serializationService,
+            this.lifecycleService,
+            this.loadBalancer,
+            this.clusterFailoverService,
+            this.failoverConfig,
+            this.clusterService,
+            this.invocationService
+        );
+
+        this.listenerService = new ListenerService(
+            this.loggingService.getLogger(),
+            this.config.network.smartRouting,
+            this.connectionManager,
+            this.invocationService
+        );
+
         this.lockReferenceIdGenerator = new LockReferenceIdGenerator();
-        this.errorFactory = new ClientErrorFactory();
-        this.statistics = new Statistics(this);
-        this.clusterViewListenerService = new ClusterViewListenerService(this);
+
+        this.proxyManager = new ProxyManager(
+            this.config,
+            this.loggingService.getLogger(),
+            this.invocationService,
+            this.listenerService,
+            this.partitionService,
+            this.serializationService,
+            this.connectionManager,
+            this.nearCacheManager,
+            this.getRepairingTask(),
+            this.clusterService,
+            this.lockReferenceIdGenerator,
+            this.getLocalClient().localAddress
```

Comment:
getLocalClient() is moved to client to remove dependency between cluster service and connection manager